internal/controller/nonadmindownloadrequest_controller.go

```diff
@@ -0,0 +1,308 @@
+			patchErr := r.patchAddErrorStatusTrueConditionBackoff(ctx, req, nacv1alpha1.ConditionNonAdminRestoreNotAvailable, "cannot get nonadminrestore")
+			logger.Error(patchErr, statusPatchErr, req.Kind, req.Name)
+			logger.Error(err, "unable to get nar", nar.Name, nar.Namespace)
+			return err
+		}
+		// check nar for nabsl
+		nabName = nar.NonAdminBackupName()
+		veleroDR.Spec.Target.Name = nar.VeleroRestoreName()
+	}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: req.Namespace, Name: nabName}, &nab); err != nil {
+		patchErr := r.patchAddErrorStatusTrueConditionBackoff(ctx, req, nacv1alpha1.ConditionNonAdminBackupNotAvailable, "cannot get nonadminbackup")
+		logger.Error(patchErr, statusPatchErr, req.Kind, req.Name)
+		logger.Error(err, "unable to get nab", nab.Name, nab.Namespace)
+		return err
+	}
+	// error if nab does not use nabsl
+	if !nab.UsesNaBSL() {
+		patchErr := r.patchAddErrorStatusTrueConditionBackoff(ctx, req, nacv1alpha1.ConditionNonAdminBackupStorageLocationNotUsed, "backup does not use nonadminbackupstoragelocation")
+		logger.Error(patchErr, statusPatchErr, req.Kind, req.Name)
+		// patch status to completed to stop processing this NADR
+		// because it is not using NonAdminBackupStorageLocation, user is expected to recreate NADR
+		// after they have a NAB using NABSL
+		prePatch := req.DeepCopy()
+		req.Status.Phase = nacv1alpha1.NonAdminPhaseCreated // not using nabsl is terminal
+		if patchErr := r.Status().Patch(ctx, req, client.MergeFrom(prePatch)); patchErr != nil {
+			logger.Error(patchErr, statusPatchErr, req.Kind, req.Name)
+			return patchErr
+		}
+		return nil
+	}
+	// if VDR has target name, it is populated by restore case
+	// if VDR do not have target name, this is a backup case
+	// so set veleroDR.Spec.Target.Name to nab.VeleroBackupName()
+	if veleroDR.Spec.Target.Name == constant.EmptyString {
+		veleroDR.Spec.Target.Name = nab.VeleroBackupName()
+	}
+	// veleroDR is now ready to be created
+	if veleroDR.ResourceVersion == constant.EmptyString {
+		if err := r.Create(ctx, veleroDR); err != nil {
+			// requeue so Get can update nadr status (if exists) or recreate
+			return err
+		}
+	}
+	return nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+// We are using predicates within For, and Watches itself, instead of defining complicated CompositePredicate that applies to all watches/fors.
+// This approach is more readable and less lines of code, and is self contained within each controller.
+// Other controllers had to build CompositePredicate because they used EventFilter which applied to all watches.
+func (r *NonAdminDownloadRequestReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&nacv1alpha1.NonAdminDownloadRequest{}, builder.WithPredicates(ctrlpredicate.Funcs{
+			CreateFunc: func(tce event.TypedCreateEvent[client.Object]) bool {
```

Comment:
we should always return true for create events

Reply:
only if CRD validation blocks all invalids.. do we want that?

Reply:
since this is the controller for this CRD, should it not process all objects creation?

Reply:
We can always return true if all creations have target/kind which it seems we do via velero crd validaiton markers. We can consider simplyfy to returing true here.